specification/dartLangSpec.tex

```diff
@@ -9168,33 +9169,56 @@ \subsection{Symbols}
 representing the reserved word \VOID.
 
 \LMHash{}%
-Assume that the term $t$ is an identifier \id{}
-that does not start with an underscore,
-that $t$ is \VOID,
-that $t$ is a period separated sequence of identifiers
-\code{$id_1$.$id_2$.$\cdots$.$id_n$},
-or that $t$ is derived from \synt{operator}.
-Any two occurrences of \code{\#t} evaluate to the same object
-(\commentary{that is, symbol instances are canonicalized}),
-and no other symbol literals evaluate to that object,
-nor to a \code{Symbol} instance that is equal to that object
-according to the \lit{==} operator
+Assume that the term $t$ is such that
+one of the following conditions is satisfied:
+
+\begin{itemize}
+\item $t$ is an identifier that does not start with an underscore.
+\item $t$ is \VOID.
+\item $t$ is a period separated sequence of identifiers
+  \code{$id_1$.$id_2$.$\cdots$.$id_n$}.
+\item $t$ is derived from \synt{operator}.
+\end{itemize}
+
+\LMHash{}%
+Any two occurrences of \code{\#t} evaluate to the same object,
+and so does a symbol obtained as the value of a constant expression
+of the form \code{Symbol($s$)} or \code{\CONST{} Symbol($s$)},
+where $s$ is a constant string whose contents is $t$.
```

Comment:
Would:
 ... a constant expression evaluating to a string ...
be more precise, or is this an imprecision we're fine with otherwise?

Also, the contents of a string is a sequence of code units, it's not a term, so maybe "whose contents is the same characters as $t$". (Very nit-picky).

Reply:
'a constant expression evaluating to a string' is surely better. I used the nit, too. ;)